src/execution_plan/execution_plan_modify.c

```diff
@@ -260,34 +270,26 @@ OpBase *ExecutionPlan_LocateOp(OpBase *root, OPType type) {
 	return NULL;
 }
 
-void ExecutionPlan_Taps(OpBase *root, OpBase ***taps) {
-	if(root == NULL) return;
-	if(root->type & OP_SCAN) *taps = array_append(*taps, root);
-
-	for(int i = 0; i < root->childCount; i++) {
-		ExecutionPlan_Taps(root->children[i], taps);
-	}
-}
-
 OpBase *ExecutionPlan_LocateReferences(OpBase *root, rax *references) {
 	OpBase *op = NULL;
 	const char **temp = _ExecutionPlan_LocateReferences(root, &op, references);
 	array_free(temp);
 	return op;
 }
 
-// Collect all resolved entities on an operation chain.
-void ExecutionPlan_ResolvedModifiers(const OpBase *op, rax *modifiers) {
+// Collect all aliases that have been resolved by the given tree of operations.
+void ExecutionPlan_BoundVariables(const OpBase *op, rax *modifiers) {
 	assert(op && modifiers);
 	if(op->modifies) {
 		uint modifies_count = array_len(op->modifies);
 		for(uint i = 0; i < modifies_count; i++) {
 			const char *modified = op->modifies[i];
-			raxTryInsert(modifiers, (unsigned char *)modified, strlen(modified), NULL, NULL);
+			raxTryInsert(modifiers, (unsigned char *)modified, strlen(modified), (void *)modified, NULL);
```

Comment:
I want to collect the const string pointers from the AST so that we can safely populate `modifies` arrays. In addition to not needing to worry about memory scoping, this reduces the number of memcpy/strdup calls we need to make in order to use keys, and lets us avoid the NULL terminator logic we regularly mess up!